docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+
+- When a user doesn’t specify either of `add_all_backend_roles` or `backend_roles` fields and set model_access_mode to _restricted_, there’ll be an exception: `You must specify one or more backend roles or add all backend roles to register a restricted model group.`
+
+- When a user specified both `add_all_backend_roles` and `backend_roles` and set model_access_mode to _restricted_, there’ll be an exception: `You cannot specify backend roles and add all backend roles at the same time`
+
+- When a user specified `backend_roles` and set model_access_mode to _restricted_, but the user doesn’t have all the backend_roles specified, there’ll be an exception: `You cannot specify backend roles and add all backend roles at the same time`
+
+### Note
+- If security or `model_access_control_enabled` setting is disabled in the cluster, every user is null and by default, in such cluster, all model groups are public. User can register a model group with a `name` and `description` but cannot specify any of the access parameters (`model_access_name`, `backend_roles`, or `add_backend_roles`). An exception will be thrown if the request has any access control related fields in the input: `You cannot specify model access control parameters because Security plugin or model access control is disabled on your cluster.`
+
+
+
+
+## Updating a model group
+
+Updating a model group request is very similar to register model group request. Additionally, user needs to specify the `model_group_id` field in the request to which he/she needs to make an update.
+
+
+### Path and HTTP method
+
+```
+PUT /_plugins/_ml/model_groups/<model_group_id>_update
+```
+
+* A user can make updates to a model group to which he/she has access which is determined by the access mode of the model group.
+* In a model_access_control enabled cluster, the model group owner or an admin user can update all fields. Whereas, any other user who shares one or more backend roles with the model group can update the `name` and `description` fields only.
+* In a security/model_access_control disabled cluster, any user can make updates to any model group but cannot specify any of the access fields.
+* Consider a model group created by User1 with IT backend_role set to it. Since user1 is the owner, he/she will be able to update all the fields of the model group.
+* If User2 tries to send update request to the same model group, only name, and description will be allowed to be updated but not other fields because user2 has access to the model group but is not the owner/admin.
+* If User3 tries to send update request, exception will be thrown because user3 has no matching backend_roles of the model group and therefore has no access to it at all.
+
+Similar to registering, there are other checks for updating a model group.
```

Comment:
These exceptions seems not the most important things for most users who are using this feature in correct way. How about we add a new section for all checkings and exceptions like register and update model group? So we can keep the doc clean and tight for most happy cases. Otherwise, user have to read through a long document to start use, they may be lost in too much details.

Reply:
Actually I have removed the exception cases from this doc. As you mentioned, this might cause unnecessary confusion to the user. Just made sure that user knows which field should be specified in what case